python/metatensor-operations/metatensor/operations/_dispatch.py

```diff
@@ -146,6 +172,23 @@ def list_to_array(array, data: Union[List[int], List[List[int]]]):
         return np.array(data)
     else:
         raise TypeError(UNKNOWN_ARRAY_TYPE)
+    
+
+def mask(array, axis: int, mask):
```

Comment:
what's the issue with `array[mask]`?

Reply:
The issue is when doing properties. In that case, you would really like to do `array[..., mask]`, but indexing after an ellipsis is not supported in torchscript

Reply:
ok, fair enough. Might be worth adding a small comment explaining that to the next person who will touch this code!

Reply:
Will add the comment